spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/AbstractAdaptableMessageListener.java

```diff
@@ -173,12 +175,21 @@ public void setMessageConverter(MessageConverter messageConverter) {
 	}
 
 	/**
-	 * Set a post processor to process the reply immediately before {@code Channel#basicPublish()}.
-	 * Often used to compress the data.
+	 * Set a post processor to process the reply immediately before
+	 * {@code Channel#basicPublish()}. Often used to compress the data.
 	 * @param replyPostProcessor the reply post processor.
+	 * @deprecated in favor of
+	 * {@link #setBeforeSendReplyPostProcessors(MessagePostProcessor...)}.
 	 */
+	@Deprecated
 	public void setReplyPostProcessor(MessagePostProcessor replyPostProcessor) {
-		this.replyPostProcessor = replyPostProcessor;
+		this.beforeSendReplyPostProcessors = new MessagePostProcessor[] { replyPostProcessor };
```

Comment:
I think we can just call `setBeforeSendReplyPostProcessors(replyPostProcessor)` from here for consistency.

And that new `setBeforeSendReplyPostProcessors()` deserves some JavaDocs.